### ISSUE TITLE
Complete matrix and parallel functions

### ISSUE THREAD

**Issue by hadley** [hadley/vctrs#9]
Vector  | Summary   | Cumulative   | Parallel | Matrix   
--------|-----------|--------------|----------|---------
`+`     | `sum`     | `cumsum`     |          | `rowSums`
`*`     | `prod`    | `cumprod`    |          |
        | `min`     | `cummin`     | `pmin`   |
        | `max`     | `cummax`     | `pmax`   |
        | `mean`    | `cummean`    |          | `rowMeans`
`&`     | `all`     | `cumall`     |          | 
`|`     | `any`     | `cumany`     |          |

**Comment by hadley** [comment]
Particularly important for standardised NA handling

**Comment by jonocarroll** [comment]
By "_complete_" do you mean this might have scope for introducing `psum`, `pprod`, `pmean`? The critical inclusion being the option of `na.rm`.

I know it can be handled somewhat by an `ifelse` or just comparison operators, but I'd also like to see a `which.pmin`/`which.pmax` for extracting out a binary selector (also handling `NA` properly).

**Comment by hadley** [comment]
Yes, that's the whole point.  I doubt I will implement any variation of which

**Comment by jonocarroll** [comment]
Ah, and only now do I see https://github.com/hadley/dplyr/issues/968 -- yep, that would be very handy. Just got stung by it again.
